crates/ruma-common/src/identifiers/matrix_uri.rs

```diff
@@ -676,7 +676,10 @@ mod tests {
             MatrixToUri::parse("https://matrix.to/#/%21ruma%3Anotareal.hs?via=notareal.hs")
                 .expect("Failed to create MatrixToUri.");
         assert_eq!(matrix_to.id(), &room_id!("!ruma:notareal.hs").into());
-        assert_eq!(matrix_to.via(), &vec![server_name!("notareal.hs").to_owned()]);
+        assert_eq!(
+            matrix_to.via(),
+            &vec![Into::<Box<ServerName>>::into(server_name!("notareal.hs"))]
```

Comment:
This seems like unnecessary noise, can you undo it? We should change types all at once so this can stay `.to_owned()`.

Reply:
Same below.

Reply:
Er... that's not gonna work if we're not going to change `matrix_to` (here) to use the owned identifiers, it's explicitly out of scope for this PR, so this was the path of least resistance.

When we change from using box to using Owned* everywhere, this could be reverted, but I think this change should stay like this.

Reply:
> Conversion of the existing codebase is out of scope for this PR

Also, this PR doesn't change `matrix_to`, it only changes the right-hand-side comparison to match as a box, as previously `to_owned` converted it to boxes, it now converts it to owned types.

There's no `PartialEq` for owned IDs to boxes, but if you want that, I could add it.

Reply:
Having looked at the entire changeset for this file again I really don't get why this change is necessary but I guess if it is then things are fine as-is.

Reply:
Hmmm, i think these test changes are avoidable, so i'll add a PartialEq for Boxes for relative backwards compatibility, sound good?

Reply:
Sure.